src/test/url-loader/package-url-resolver_test.ts

```diff
@@ -152,6 +152,27 @@ suite('PackageUrlResolver', function() {
       assert.equal(
           resolver.resolve(fileRelativeUrl`%><><%=`, packageRoot), undefined);
     });
+
+    test('resolves a URL with no pathname', () => {
+      const foo = rootedFileUrl`1/2/foo.html?baz#bat`;
+      const bar = rootedFileUrl`1/2/bar.html`;
+      assert.equal(
+          resolver.resolve(fileRelativeUrl``, foo),
+          rootedFileUrl`1/2/foo.html?baz`);
+      assert.equal(resolver.resolve(fileRelativeUrl``, bar), bar);
+      assert.equal(
+          resolver.resolve(fileRelativeUrl`#buz`, foo),
+          rootedFileUrl`1/2/foo.html?baz#buz`);
```

Comment:
I'm pretty sure `#buz` should not be in the result

Reply:
`#buz` is in the result *because* it was in the url to resolve and because that's part of the main purpose of this PR :)